diff --git a/bin/phing.php b/bin/phing.php
index e414c24..d696230 100644
--- a/bin/phing.php
+++ b/bin/phing.php
@@ -8,10 +8,14 @@
  */
 
 // Use composers autoload.php if available
-if (file_exists(dirname(__FILE__) . '/../vendor/autoload.php')) {
-    require_once dirname(__FILE__) . '/../vendor/autoload.php';
-} elseif (file_exists(dirname(__FILE__) . '/../../../autoload.php')) {
-    require_once dirname(__FILE__) . '/../../../autoload.php';
+if (is_file($autoload = getcwd() . '/vendor/autoload.php')) {
+    require_once $autoload;
+}
+
+if (is_file($autoload = __DIR__ . '/../vendor/autoload.php')) {
+    require_once $autoload;
+} elseif (is_file($autoload = __DIR__ . '/../../../autoload.php')) {
+    require_once $autoload;
 }
 
 // Set any INI options for PHP
