diff --git a/src/Task.php b/src/Task.php
index 7fae6bb..f1da944 100644
--- a/src/Task.php
+++ b/src/Task.php
@@ -208,7 +208,11 @@ class Task extends \ExecTask {
    *   The color option.
    */
   public function setColor($yesNo) {
-    if (!$yesNo) {
+    if ($yesNo) {
+      $this->createOption()
+        ->setName('color');
+    }
+    else {
       $this->createOption()
         ->setName('nocolor');
     }
