diff --git a/CodeSniffer/File.php b/CodeSniffer/File.php
index 62c39799f..17f155b67 100644
--- a/CodeSniffer/File.php
+++ b/CodeSniffer/File.php
@@ -1763,7 +1763,7 @@ class PHP_CodeSniffer_File
                 }
                 break;
             default:
-                continue;
+                continue 2;
             }//end switch
         }//end for
 
@@ -2880,7 +2880,7 @@ class PHP_CodeSniffer_File
                 // If it's null, then there must be no parameters for this
                 // method.
                 if ($currVar === null) {
-                    continue;
+                    continue 2;
                 }
 
                 $vars[$paramCount]            = array();
diff --git a/CodeSniffer/Tokenizers/CSS.php b/CodeSniffer/Tokenizers/CSS.php
index 8e4ba35b0..8656e058b 100644
--- a/CodeSniffer/Tokenizers/CSS.php
+++ b/CodeSniffer/Tokenizers/CSS.php
@@ -390,7 +390,7 @@ class PHP_CodeSniffer_Tokenizers_CSS extends PHP_CodeSniffer_Tokenizers_PHP
 
                     // Needs to be in the format "url(" for it to be a URL.
                     if ($finalTokens[$x]['code'] !== T_OPEN_PARENTHESIS) {
-                        continue;
+                        continue 2;
                     }
 
                     // Make sure the content isn't empty.
@@ -401,7 +401,7 @@ class PHP_CodeSniffer_Tokenizers_CSS extends PHP_CodeSniffer_Tokenizers_PHP
                     }
 
                     if ($finalTokens[$y]['code'] === T_CLOSE_PARENTHESIS) {
-                        continue;
+                        continue 2;
                     }
 
                     if (PHP_CODESNIFFER_VERBOSITY > 1) {
