diff --git a/src/LEtudiant/Composer/Installer/Solver/SharedPackageSolver.php b/src/LEtudiant/Composer/Installer/Solver/SharedPackageSolver.php
index f719893..b7972e6 100644
--- a/src/LEtudiant/Composer/Installer/Solver/SharedPackageSolver.php
+++ b/src/LEtudiant/Composer/Installer/Solver/SharedPackageSolver.php
@@ -57,15 +57,16 @@ class SharedPackageSolver
     public function isSharedPackage(PackageInterface $package)
     {
         $prettyName = $package->getPrettyName();
+        $packageType = $package->getType();
 
         // Avoid putting this package into dependencies folder, because on the first installation the package won't be
         // installed in dependencies folder but in the vendor folder.
         // So I prefer keeping this behavior for further installs.
-        if (SharedPackageInstaller::PACKAGE_PRETTY_NAME === $prettyName) {
+        if (SharedPackageInstaller::PACKAGE_PRETTY_NAME === $prettyName || $packageType === 'composer-plugin') {
             return false;
         }
 
-        if ($this->areAllShared || SharedPackageInstaller::PACKAGE_TYPE === $package->getType()) {
+        if ($this->areAllShared || SharedPackageInstaller::PACKAGE_TYPE === $packageType) {
             return true;
         }
 
