diff --git a/src/Commands/AbstractDrupalCommands.php b/src/Commands/AbstractDrupalCommands.php
index 05e7e34..ce95af1 100644
--- a/src/Commands/AbstractDrupalCommands.php
+++ b/src/Commands/AbstractDrupalCommands.php
@@ -418,12 +418,12 @@ abstract class AbstractDrupalCommands extends AbstractCommands implements Filesy
         // Define collection of tasks.
         $collection = [
             // Note that the chmod() method takes decimal values.
-            $this->taskFilesystemStack()->chmod($subdirPath, octdec(775), 0000, true),
+            $this->taskFilesystemStack()->chmod($subdirPath, octdec('775'), 0000, true),
         ];
 
         if (file_exists($subdirPath . '/settings.php')) {
             // Note that the chmod() method takes decimal values.
-            $collection[] = $this->taskFilesystemStack()->chmod($subdirPath . '/settings.php', octdec(664));
+            $collection[] = $this->taskFilesystemStack()->chmod($subdirPath . '/settings.php', octdec('664'));
         }
 
         return $this->collectionBuilder()->addTaskList($collection);
