diff --git a/features/messages.feature b/features/messages.feature
index 802bbd7..0d5e136 100644
--- a/features/messages.feature
+++ b/features/messages.feature
@@ -16,4 +16,5 @@ Feature: Ensure that messages are working properly on local installs
     When I fill in "a fake user" for "Username"
     And I fill in "a fake password" for "Password"
     When I press "Log in"
+    And I wait for 5 seconds
     Then I should see the error message "Unrecognized username or password"
